r1cs/gadgets/std/src/fields/nonnative/params.rs

```diff
@@ -1,6 +1,7 @@
 use crate::fields::nonnative::NonNativeFieldParams;
 
 /// Obtain the parameters from a `ConstraintSystem`'s cache or generate a new one
+/// CAUTION: for now, it just generates new ones.
```

Comment:
In reality is a constant function, meaning that itss return value is evaluated directly at compile time. So there shouldn't been any recomputation and overhead at run-time